linked_list/linked_list.py

```diff
@@ -86,17 +148,52 @@ def visit(self):
 
     # method to reverse the singly linked list
     # note: the nodes should be moved and not just the values in the nodes
-    # Time Complexity: ?
-    # Space Complexity: ?
+    # Time Complexity: O(n^2)
+    # Space Complexity: O(1)
     def reverse(self):

```

Comment:
👍 This definitely works and I like the helper function, but it can be done in O(n) time complexity.